### PR TITLE
add: setEngine

### PR DESCRIPTION
**Pull request by zhuangya**
shorthand for set each route's engine.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -100,6 +100,70 @@ describe('Paloma test', function () {
...
+    });
+
+    request(app.callback())
+      .get('/general')
+      .expect(200)
+      .end((err, res) => {
+        if (err) return done(err);
+        assert.equal(res.type, 'text/html');
+        assert.equal(res.text, '<h1>index body</h1>');
+        request(app.callback())
```
Comment (zhuangya) [inline test/index.js]: any better way? maybe `co-parallel`?

**Comment by nswbmw** [comment]
I prefer `app.engine = 'jade'`, not change anything, maybe work :smile:

**Comment by zhuangya** [comment]
> I prefer app.engine = 'jade', not change anything, maybe work

implemented

File: lib/paloma.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ module.exports = class Paloma extends Koa {
   }
 
   route(route) {
+    route.engine = route.engine || this.engine || undefined;
```
Comment (nswbmw) [inline lib/paloma.js]: I think this line should be moved to https://github.com/palomajs/paloma-router/blob/master/lib%2Findex.js#L13

Comment (zhuangya) [inline lib/paloma.js]: but there is no `app` in `paloma-router` :(

Comment (nswbmw) [inline lib/paloma.js]: `this` is bind to `app` :)

**Comment by zhuangya** [comment]
done in palomajs/paloma-router#2

**Pull request closed without merging**